### PR TITLE
support single member annotation

### PR DESCRIPTION
**Pull request by Wreulicke**
ref #1484

* only "value" member Annotation converts SingleMemberAnnotation in pretty print.
* fix tests

### CODE REVIEW

File: src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java [hunk 1]
```diff
@@ -374,9 +374,15 @@ public String toString() {
...
+		if (size == 1 && values.containsKey("value")) {
+			printer.write('(');
+			elementPrinterHelper.writeAnnotationElement(annotation.getFactory(), annotation.getValues().get("value"));
+			printer.write(')');
+		} else if (size > 0) {
 			try (ListPrinter lp = printer.createListPrinter("(", ", ", ")")) {
-				for (Entry<String, CtExpression> e : annotation.getValues().entrySet()) {
+				for (Entry<String, CtExpression> e : values.entrySet()) {
 					lp.printSeparatorIfAppropriate();
 					printer.write(e.getKey() + " = ");
```
Comment (pvojtechovsky) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: Thank You for this PR!
What about simply change only this line like:
```java
if ((annotation.getValues().size() == 1 && "value".equals(e.key())) == false) {
   //it is not a default value attribute. We must print a attribute name too.
   printer.write(e.getKey() + " = ");
}
```
WDYT?

Comment (Wreulicke) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: good.
I will reflect your opinion to this PR.

**Comment by monperrus** [comment]
Hi @Wreulicke, thanks a lot for your PR, it looks really good to me.

are you sure this is 100% compatible? does the Java language specification state something about this equivalence?

**Comment by pvojtechovsky** [comment]
> does the Java language specification state something about this equivalence?

There is following chapter [9.7.3. Single-Element Annotations](http://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.7.3)

**Comment by monperrus** [comment]
thanks a lot!

**Pull request merged**